server.go

```diff
@@ -0,0 +1,75 @@
+package unixtransport
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+)
+
+// A TestServer is an HTTP+UNIX server listening on a temporary UNIX socket, for
+// use in end-to-end HTTP tests. TestServer's API mimics that of the
+// net/http/httptest.Server type.
+type TestServer struct {
+	// Base URL of form http+unix:///tmp/(random)/test.sock: with no trailing
+	// slash.
+	URL string
+
+	server *httptest.Server
+	client *http.Client
+
+	// Temporary directory, cleaned up on close.
+	dir string
+}
+
+// NewTestServer starts and returns a new TestServer. The caller should call
+// Close when finished, to shut it down.
+func NewTestServer(handler http.Handler) *TestServer {
+	// If any subsequent calls fail before we return *TestServer, be sure to
+	// os.RemoveAll(dir) to tidy things up.
+	dir, err := os.MkdirTemp(os.TempDir(), "unixtransport-server")
+	if err != nil {
+		panicf("failed to make temporary directory: %v", err)
+	}
+
```

Comment:
`defer os.RemoveAll(...)`?

Reply:
I don't believe so: right now we generate the temporary directory and either remove it only on error returns, or on a call to Close. I don't know what happens if we have a UNIX socket listener open but then remove the directory it exists in before returning to the caller.

If we change the API to take a `testing.TB` though, I can use `tb.TempDir()` and let the test clean it all up at the end. That is simpler and easier to reason about too.